### ISSUE TITLE
no love for the gov

### ISSUE THREAD

**Issue by mrgleeco** [mozilla/sops#214]
i've great success with running sops in a typical aws account.  However, when i run in us-gov-west-1, there is always an error:

```
Could not load file: Error encrypting the data key with one or more master keys: [Failed to encrypt new data key with master key "<redacted  keyname>": Failed to create session: No valid ARN found in "<redacted keyname>"]
```

it looks as if the `keysource.go` is hardcoded regex for `^arn:aws:kms:`.
But the arn string for govcloud has a pattern like `arn:aws-us-gov:kms:us-gov-west-1:1234567890:key/...`

**Comment by autrilla** [comment]
Thanks for the bug report.

According to [Amazon](http://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html#genref-arns) the ARN formats are:

```
arn:partition:service:region:account-id:resource
arn:partition:service:region:account-id:resourcetype/resource
arn:partition:service:region:account-id:resourcetype:resource
```

With `partition` being `aws-partitionname` or `aws`.

So I suppose we could just change the regex to `^arn:aws[\w-]+:kms:(.+):([0-9]+):key/(.+)$`. Ideally we'd find some sort of function in the AWS SDK that lets us extract the region and role from the ARN.

**Comment by mrgleeco** [comment]
sdk looks to define (& export) at https://github.com/aws/aws-sdk-go/blob/master/aws/endpoints/defaults.go#L11-L13

possibly a regex with a test to string match one of these?  Feels brittle, but also likely not too much changes here.

**Comment by mrgleeco** [comment]
ping -
and thx @autrilla for looking

i've a patch in branch at https://github.com/gleeco/sops/commit/b715e29f6b821920953d3d7c1c5bdef10e7aab53

i've not a testing story;  there looks like special hooks wrt `createSession` which may mean insane mockery;  any guidance from the team?

thx in advance!

**Comment by autrilla** [comment]
This has been fixed

**Comment by mrgleeco** [comment]
👍  - is there a release schedule?

**Comment by jvehent** [comment]
I just released 2.0.10 https://github.com/mozilla/sops/releases/tag/2.0.10
